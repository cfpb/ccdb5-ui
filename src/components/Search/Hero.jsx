import './Hero.less';
import { MODAL_SHOWN, MODAL_TYPE_MORE_ABOUT } from '../../constants'
import { connect } from 'react-redux'
import React from 'react';

/**
* Notifies the application that the "More About..." dialog box should appear
*
* @returns {string} a packaged payload to be used by Redux reducers
*/
export function showMoreAboutDialog() {
  return {
    type: MODAL_SHOWN,
    modalType: MODAL_TYPE_MORE_ABOUT,
    modalProps: {}
  }
}

export class Hero extends React.Component {
  render() {
    return (
        <header className="content_hero">
          <h1 className="content-header">Consumer Complaint Database</h1>
          <ul className="m-list m-list__horizontal">
            <li className="m-list_item">
              <button className="a-btn a-btn__link"
                      onClick={this.props.onMoreAbout}>
                Things to know before you use this database
              </button>
            </li>
            <li className="m-list_item">
              <a href="https://www.consumerfinance.gov/complaint/data-use/"
                 target="_blank"
                 rel="noopener noreferrer">
                How we use complaint data
              </a>
            </li>
            <li className="m-list_item">
              <a href="https://cfpb.github.io/api/ccdb/"
                 target="_blank"
                 rel="noopener noreferrer">
                Technical documentation
              </a>
            </li>
          </ul>
        </header>
    );
  }
}

export const mapDispatchToProps = dispatch => ( {
  onMoreAbout: () => {
    dispatch( showMoreAboutDialog() )
  }
} )

export default connect( null, mapDispatchToProps )( Hero )
