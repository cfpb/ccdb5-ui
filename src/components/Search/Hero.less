@import (less) '../../css/base.less';

.content_hero {
  padding: @gutter-normal;

  @media @desktop {
    padding: @grid_gutter-width;
    padding-top: @gutter-normal;
  }

  @media @phone {
    padding: @gutter-narrow;
  }

  margin-top: @grid_gutter-width;

  p {
    max-width: 670px;
  }

  ul {
    padding-top: @gutter-minimum;
  }

  li {
    padding-bottom: 0;
    border-bottom: 0;

    button, a {
      font-size: 16px;
    }

    &.m-list_item {
      margin-right: 0;
    }

    &::after {
      margin-left: @gutter-narrow;
      margin-right: @gutter-narrow;

      content: '|';
    }

    &:last-of-type {
      &::after {
        margin: 0;

        content: '';
      }
    }
  }

  .content-header {
    display:inline-flex;
  }
  h4.content-header {
    box-sizing: border-box;
    margin-left: @gutter-narrow;
    margin-top: 0;
    color: @dark-gray;
    text-transform: uppercase;
    vertical-align: text-bottom;
  }
}
