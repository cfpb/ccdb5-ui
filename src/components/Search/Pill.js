import './Pill.less';
import {
  removeFilter,
  changeDateRange,
  replaceFilters,
} from '../../reducers/query/query';
import { filterPatch, SLUG_SEPARATOR } from '../../constants';
import { formatPillPrefix, getUpdatedFilters } from '../../utils/filters';
import { useDispatch, useSelector } from 'react-redux';
import { coalesce } from '../../utils';
import iconMap from '../iconMap';
import PropTypes from 'prop-types';
import React from 'react';
import { selectAggsState } from '../../reducers/aggs/selectors';
import { selectQueryState } from '../../reducers/query/selectors';

export const Pill = ({ fieldName, value }) => {
  const aggsState = useSelector(selectAggsState);
  const queryState = useSelector(selectQueryState);
  const aggs = coalesce(aggsState, fieldName, []);
  const filters = coalesce(queryState, fieldName, []);
  const prefix = formatPillPrefix(fieldName);
  const trimmed = value.split(SLUG_SEPARATOR).pop();
  const dispatch = useDispatch();

  const remove = () => {
    if (fieldName === 'date_received') {
      // reset date range
      dispatch(changeDateRange('All'));
    } else {
      const filterName = value;
      if (filterPatch.includes(fieldName)) {
        const updatedFilters = getUpdatedFilters(
          filterName,
          filters,
          aggs,
          fieldName
        );
        dispatch(replaceFilters(fieldName, updatedFilters));
      } else {
        dispatch(
          removeFilter({ fieldName: fieldName, fieldValue: filterName })
        );
      }
    }
  };

  return (
    <li>
      <button className="pill flex-fixed" onClick={remove}>
        <span className="name">
          {prefix}
          {trimmed}
        </span>
        <span className="u-visually-hidden">
          {`Remove ${trimmed} as a filter`}
        </span>
        {iconMap.getIcon('delete')}
      </button>
    </li>
  );
};

Pill.propTypes = {
  fieldName: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
};
