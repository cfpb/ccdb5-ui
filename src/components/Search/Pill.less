.pill {
  padding: @grid_gutter-width / 3 30px @grid_gutter-width / 3 @grid_gutter-width /
    3;
  border: 1px solid @teal-60;
  margin-right: 4px;
  border-radius: 0.28571429em;
  background-color: @teal-40;
  font-weight: 500;
  color: @black;
  font-size: @size-v;
  &:hover {
    background-color: #c4dddc;
    border-color: #9ec4c3;
    cursor: pointer;
  }
  // Close button
  button {
    padding: 0;
    padding-left: 5px;
    border: 0;
    border-left: 1px solid @gray-20;
    margin: 0;
    background-color: transparent;
    color: @teal-dark;
    position: absolute;
    right: 10px;
    top: 50%;
    transform: translateY(-50%);
    &:hover,
    &:active {
      background-color: @gray-40;
    }
  }
}
