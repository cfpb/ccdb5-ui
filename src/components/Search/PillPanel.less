.pill-panel {
  padding-right: @gutter-narrow;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: baseline;
  .layout-row {
    display: block;
  }
  .clear-all {
    button {
      border-color: var(--red-dark);
      color: var(--red-dark);
      margin-left: @gutter-narrow;
      svg {
        margin-right: @gutter-narrow / 2;
      }
    }
  }

  .pill-label {
    padding-right: @gutter-normal;
    margin-bottom: 0;
  }

  li {
    display: inline-block;
    position: relative;
  }

  ul {
    padding-top: @gutter-narrow;
  }

  @media @phone {
    flex-direction: column;
    margin-top: 10px;
    .layout-row {
      padding-left: 0;
    }
    .clear-all {
      display: block;
      button {
        margin: 0;
      }
    }
  }
}
