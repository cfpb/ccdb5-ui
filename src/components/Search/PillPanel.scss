@use 'sass:math';
@use 'css/base' as *;

.pill-panel {
  padding-right: $gutter-narrow;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: baseline;
  .layout-row {
    display: block;
  }
  .clear-all {
    button {
      border-color: var(--red-dark);
      color: var(--red-dark);
      margin-left: $gutter-narrow;
      svg {
        margin-right: math.div($gutter-narrow, 2);
      }
    }
  }

  .pill-label {
    padding-right: $gutter-normal;
    margin-bottom: 0;
  }

  li {
    display: inline-block;
    position: relative;
  }

  ul {
    padding-top: $gutter-narrow;
  }

  @media screen and (max-width: $layout-breakpoint-phone-max) {
    flex-direction: column;
    margin-top: 70px;
    .layout-row {
      padding-left: 0;
    }
    .clear-all {
      display: block;
      button {
        margin: 0;
      }
    }
  }
}
