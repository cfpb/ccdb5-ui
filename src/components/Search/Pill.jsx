import './Pill.scss';
import { dateRangeChanged } from '../../reducers/query/querySlice';
import {
  filterRemoved,
  filtersReplaced,
} from '../../reducers/filters/filtersSlice';
import { filterPatch, SLUG_SEPARATOR } from '../../constants';
import { formatPillPrefix, getUpdatedFilters } from '../../utils/filters';
import { useDispatch, useSelector } from 'react-redux';
import { coalesce } from '../../utils';
import getIcon from '../Common/Icon/iconMap';
import PropTypes from 'prop-types';
import { selectFiltersRoot } from '../../reducers/filters/selectors';
import { useGetAggregations } from '../../api/hooks/useGetAggregations';

export const Pill = ({ fieldName, value }) => {
  const { data: aggsState } = useGetAggregations();
  const filtersState = useSelector(selectFiltersRoot);
  const aggs = coalesce(aggsState, fieldName, []);
  const filters = coalesce(filtersState, fieldName, []);
  const prefix = formatPillPrefix(fieldName);
  const trimmed = value.split(SLUG_SEPARATOR).pop();
  const dispatch = useDispatch();

  const remove = () => {
    if (fieldName === 'date_received') {
      // reset date range
      dispatch(dateRangeChanged('All'));
    } else {
      const filterName = value;
      if (filterPatch.includes(fieldName)) {
        const updatedFilters = getUpdatedFilters(
          filterName,
          filters,
          aggs,
          fieldName,
        );
        dispatch(filtersReplaced(fieldName, updatedFilters));
      } else {
        dispatch(filterRemoved(fieldName, filterName));
      }
    }
  };

  return (
    <li>
      <button className="pill flex-fixed" onClick={remove}>
        <span className="name">
          {prefix}
          {trimmed}
        </span>
        {getIcon('delete')}
      </button>
    </li>
  );
};

Pill.propTypes = {
  fieldName: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
};
