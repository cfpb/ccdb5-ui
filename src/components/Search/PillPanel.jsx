import './PillPanel.scss';
import { DATE_RANGE_MIN, knownFilters } from '../../constants';

import {
  selectFiltersHasNarrative,
  selectFiltersRoot,
} from '../../reducers/filters/selectors';
import {
  selectQueryDateReceivedMax,
  selectQueryDateReceivedMin,
  selectQuerySearchField,
} from '../../reducers/query/selectors';

import { useDispatch, useSelector } from 'react-redux';
import dayjs from 'dayjs';
import getIcon from '../Common/Icon/iconMap';
import { Pill } from './Pill';
import { filtersCleared } from '../../reducers/filters/filtersSlice';
import { startOfToday } from '../../utils';

/* eslint complexity: ["error", 5] */
export const PillPanel = () => {
  const dispatch = useDispatch();
  const filterState = useSelector(selectFiltersRoot);
  const hasNarrative = useSelector(selectFiltersHasNarrative);

  const dateReceivedMin = useSelector(selectQueryDateReceivedMin);
  const dateReceivedMax = useSelector(selectQueryDateReceivedMax);
  const searchField = useSelector(selectQuerySearchField);

  const filters = knownFilters
    // Only use the known filters that are in the query
    .filter((filter) => filter in filterState)
    // Create a flattened array of pill objects
    .reduce((accum, fieldName) => {
      const arr = filterState[fieldName].map((value) => ({ fieldName, value }));
      return accum.concat(arr);
    }, []);

  // Add Has Narrative, if it exists
  if (hasNarrative) {
    filters.push({
      fieldName: 'has_narrative',
      value: 'Has narrative',
    });
  }

  // only add the filter the date is NOT the "All"
  if (
    !dayjs(dateReceivedMin).isSame(dayjs(DATE_RANGE_MIN), 'day') ||
    !dayjs(dateReceivedMax).isSame(dayjs(startOfToday()), 'day')
  ) {
    filters.unshift({
      fieldName: 'date_received',
      value:
        'Date Received: ' +
        dayjs(dateReceivedMin).format('M/D/YYYY') +
        ' - ' +
        dayjs(dateReceivedMax).format('M/D/YYYY'),
    });
  }

  if (!filters.length) {
    return null;
  }

  return (
    <section className="pill-panel">
      <h3 className="h4 pill-label flex-fixed">Filters applied:</h3>
      <ul className="layout-row">
        {filters.map((filter) => (
          <Pill
            key={filter.fieldName + filter.value}
            fieldName={filter.fieldName}
            value={filter.value}
          />
        ))}
        <li className="clear-all">
          <button
            className="a-btn a-btn--link body-copy"
            onClick={() => dispatch(filtersCleared(searchField))}
          >
            {getIcon('delete')}
            Clear all filters
          </button>
        </li>
      </ul>
    </section>
  );
};
