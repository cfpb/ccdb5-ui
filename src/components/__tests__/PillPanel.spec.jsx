import React from 'react'
import { PillPanel } from '../Search/PillPanel'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import renderer from 'react-test-renderer'


function setupSnapshot( initialQueryState = {}, initialAggState = {} ) {
  const middlewares = [thunk]
  const mockStore = configureMockStore(middlewares)
  const store = mockStore({
    aggs: initialAggState,
    query: initialQueryState
  })

  return renderer.create(
    <Provider store={store}>
      <PillPanel />
    </Provider>
  )
}

describe('component:PillPanel', () => {
  it('renders without crashing', () => {
    const target = setupSnapshot({
      company: ['Apples', 'Bananas are great'],
      timely: ['Yes']
    });
    const tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it( 'does not renders patched filters', () => {
    const aggs = {
      issue: [
        {
          key: 'a',
          'sub_issue.raw': {
            buckets: [ { key: 'b' }, { key: 'c' }, { key: 'd' } ]
          }
        }
      ]
    }
    const target = setupSnapshot( {
      issue: [ 'a', 'Bananas are great' ]
    }, aggs )
    const tree = target.toJSON()
    expect( tree ).toMatchSnapshot()
  } )

  it('does not render when there are no filters', () => {
    const target = setupSnapshot();
    const tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('adds a has narrative pill', () => {
    const target = setupSnapshot({
      has_narrative: true
    });
    const tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  })

  // TODO: rewrite tests for redux actions

})
