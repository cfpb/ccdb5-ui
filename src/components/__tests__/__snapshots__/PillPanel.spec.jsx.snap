// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component:PillPanel adds a has narrative pill 1`] = `
<section
  className="pill-panel"
>
  <h3
    className="h4 pill-label flex-fixed"
  >
    Filters applied:
  </h3>
  <ul
    className="layout-row"
  >
    <li
      className="pill flex-fixed"
    >
      <span
        className="name"
      >
        
        Has narrative
      </span>
      <button
        onClick={[Function]}
      >
        <span
          className="u-visually-hidden"
        >
          Remove Has narrative as a filter
        </span>
        <svg
          className="cf-icon-svg"
          viewBox="0 0 718.9 1200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M451.4 613.7l248.1-248.1c25.6-25.1 26-66.3.8-91.9s-66.3-26-91.9-.8l-.8.8-248.1 248.1-248.1-248.1c-25.4-25.4-66.5-25.4-91.9 0s-25.4 66.5 0 91.9l248.1 248.1L19.5 861.8c-25.6 25.1-26 66.3-.8 91.9s66.3 26 91.9.8l.8-.8 248.1-248.1 248.1 248.1c25.4 25.4 66.5 25.4 91.9 0s25.4-66.5 0-91.9L451.4 613.7z"
          />
        </svg>
      </button>
    </li>
    <li
      className="clear-all"
    >
      <button
        className="a-btn a-btn__link body-copy"
        onClick={[Function]}
      >
        <svg
          className="cf-icon-svg"
          viewBox="0 0 718.9 1200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M451.4 613.7l248.1-248.1c25.6-25.1 26-66.3.8-91.9s-66.3-26-91.9-.8l-.8.8-248.1 248.1-248.1-248.1c-25.4-25.4-66.5-25.4-91.9 0s-25.4 66.5 0 91.9l248.1 248.1L19.5 861.8c-25.6 25.1-26 66.3-.8 91.9s66.3 26 91.9.8l.8-.8 248.1-248.1 248.1 248.1c25.4 25.4 66.5 25.4 91.9 0s25.4-66.5 0-91.9L451.4 613.7z"
          />
        </svg>
        Clear all filters
      </button>
    </li>
  </ul>
</section>
`;

exports[`component:PillPanel does not render when there are no filters 1`] = `null`;

exports[`component:PillPanel does not renders patched filters 1`] = `
<section
  className="pill-panel"
>
  <h3
    className="h4 pill-label flex-fixed"
  >
    Filters applied:
  </h3>
  <ul
    className="layout-row"
  >
    <li
      className="pill flex-fixed"
    >
      <span
        className="name"
      >
        
        a
      </span>
      <button
        onClick={[Function]}
      >
        <span
          className="u-visually-hidden"
        >
          Remove a as a filter
        </span>
        <svg
          className="cf-icon-svg"
          viewBox="0 0 718.9 1200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M451.4 613.7l248.1-248.1c25.6-25.1 26-66.3.8-91.9s-66.3-26-91.9-.8l-.8.8-248.1 248.1-248.1-248.1c-25.4-25.4-66.5-25.4-91.9 0s-25.4 66.5 0 91.9l248.1 248.1L19.5 861.8c-25.6 25.1-26 66.3-.8 91.9s66.3 26 91.9.8l.8-.8 248.1-248.1 248.1 248.1c25.4 25.4 66.5 25.4 91.9 0s25.4-66.5 0-91.9L451.4 613.7z"
          />
        </svg>
      </button>
    </li>
    <li
      className="pill flex-fixed"
    >
      <span
        className="name"
      >
        
        Bananas are great
      </span>
      <button
        onClick={[Function]}
      >
        <span
          className="u-visually-hidden"
        >
          Remove Bananas are great as a filter
        </span>
        <svg
          className="cf-icon-svg"
          viewBox="0 0 718.9 1200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M451.4 613.7l248.1-248.1c25.6-25.1 26-66.3.8-91.9s-66.3-26-91.9-.8l-.8.8-248.1 248.1-248.1-248.1c-25.4-25.4-66.5-25.4-91.9 0s-25.4 66.5 0 91.9l248.1 248.1L19.5 861.8c-25.6 25.1-26 66.3-.8 91.9s66.3 26 91.9.8l.8-.8 248.1-248.1 248.1 248.1c25.4 25.4 66.5 25.4 91.9 0s25.4-66.5 0-91.9L451.4 613.7z"
          />
        </svg>
      </button>
    </li>
    <li
      className="clear-all"
    >
      <button
        className="a-btn a-btn__link body-copy"
        onClick={[Function]}
      >
        <svg
          className="cf-icon-svg"
          viewBox="0 0 718.9 1200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M451.4 613.7l248.1-248.1c25.6-25.1 26-66.3.8-91.9s-66.3-26-91.9-.8l-.8.8-248.1 248.1-248.1-248.1c-25.4-25.4-66.5-25.4-91.9 0s-25.4 66.5 0 91.9l248.1 248.1L19.5 861.8c-25.6 25.1-26 66.3-.8 91.9s66.3 26 91.9.8l.8-.8 248.1-248.1 248.1 248.1c25.4 25.4 66.5 25.4 91.9 0s25.4-66.5 0-91.9L451.4 613.7z"
          />
        </svg>
        Clear all filters
      </button>
    </li>
  </ul>
</section>
`;

exports[`component:PillPanel renders without crashing 1`] = `
<section
  className="pill-panel"
>
  <h3
    className="h4 pill-label flex-fixed"
  >
    Filters applied:
  </h3>
  <ul
    className="layout-row"
  >
    <li
      className="pill flex-fixed"
    >
      <span
        className="name"
      >
        
        Date Received: 12/1/2011 - 12/1/2019
      </span>
      <button
        onClick={[Function]}
      >
        <span
          className="u-visually-hidden"
        >
          Remove Date Received: 12/1/2011 - 12/1/2019 as a filter
        </span>
        <svg
          className="cf-icon-svg"
          viewBox="0 0 718.9 1200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M451.4 613.7l248.1-248.1c25.6-25.1 26-66.3.8-91.9s-66.3-26-91.9-.8l-.8.8-248.1 248.1-248.1-248.1c-25.4-25.4-66.5-25.4-91.9 0s-25.4 66.5 0 91.9l248.1 248.1L19.5 861.8c-25.6 25.1-26 66.3-.8 91.9s66.3 26 91.9.8l.8-.8 248.1-248.1 248.1 248.1c25.4 25.4 66.5 25.4 91.9 0s25.4-66.5 0-91.9L451.4 613.7z"
          />
        </svg>
      </button>
    </li>
    <li
      className="pill flex-fixed"
    >
      <span
        className="name"
      >
        
        Apples
      </span>
      <button
        onClick={[Function]}
      >
        <span
          className="u-visually-hidden"
        >
          Remove Apples as a filter
        </span>
        <svg
          className="cf-icon-svg"
          viewBox="0 0 718.9 1200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M451.4 613.7l248.1-248.1c25.6-25.1 26-66.3.8-91.9s-66.3-26-91.9-.8l-.8.8-248.1 248.1-248.1-248.1c-25.4-25.4-66.5-25.4-91.9 0s-25.4 66.5 0 91.9l248.1 248.1L19.5 861.8c-25.6 25.1-26 66.3-.8 91.9s66.3 26 91.9.8l.8-.8 248.1-248.1 248.1 248.1c25.4 25.4 66.5 25.4 91.9 0s25.4-66.5 0-91.9L451.4 613.7z"
          />
        </svg>
      </button>
    </li>
    <li
      className="pill flex-fixed"
    >
      <span
        className="name"
      >
        
        Bananas are great
      </span>
      <button
        onClick={[Function]}
      >
        <span
          className="u-visually-hidden"
        >
          Remove Bananas are great as a filter
        </span>
        <svg
          className="cf-icon-svg"
          viewBox="0 0 718.9 1200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M451.4 613.7l248.1-248.1c25.6-25.1 26-66.3.8-91.9s-66.3-26-91.9-.8l-.8.8-248.1 248.1-248.1-248.1c-25.4-25.4-66.5-25.4-91.9 0s-25.4 66.5 0 91.9l248.1 248.1L19.5 861.8c-25.6 25.1-26 66.3-.8 91.9s66.3 26 91.9.8l.8-.8 248.1-248.1 248.1 248.1c25.4 25.4 66.5 25.4 91.9 0s25.4-66.5 0-91.9L451.4 613.7z"
          />
        </svg>
      </button>
    </li>
    <li
      className="pill flex-fixed"
    >
      <span
        className="name"
      >
        Timely: 
        Yes
      </span>
      <button
        onClick={[Function]}
      >
        <span
          className="u-visually-hidden"
        >
          Remove Yes as a filter
        </span>
        <svg
          className="cf-icon-svg"
          viewBox="0 0 718.9 1200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M451.4 613.7l248.1-248.1c25.6-25.1 26-66.3.8-91.9s-66.3-26-91.9-.8l-.8.8-248.1 248.1-248.1-248.1c-25.4-25.4-66.5-25.4-91.9 0s-25.4 66.5 0 91.9l248.1 248.1L19.5 861.8c-25.6 25.1-26 66.3-.8 91.9s66.3 26 91.9.8l.8-.8 248.1-248.1 248.1 248.1c25.4 25.4 66.5 25.4 91.9 0s25.4-66.5 0-91.9L451.4 613.7z"
          />
        </svg>
      </button>
    </li>
    <li
      className="clear-all"
    >
      <button
        className="a-btn a-btn__link body-copy"
        onClick={[Function]}
      >
        <svg
          className="cf-icon-svg"
          viewBox="0 0 718.9 1200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M451.4 613.7l248.1-248.1c25.6-25.1 26-66.3.8-91.9s-66.3-26-91.9-.8l-.8.8-248.1 248.1-248.1-248.1c-25.4-25.4-66.5-25.4-91.9 0s-25.4 66.5 0 91.9l248.1 248.1L19.5 861.8c-25.6 25.1-26 66.3-.8 91.9s66.3 26 91.9.8l.8-.8 248.1-248.1 248.1 248.1c25.4 25.4 66.5 25.4 91.9 0s25.4-66.5 0-91.9L451.4 613.7z"
          />
        </svg>
        Clear all filters
      </button>
    </li>
  </ul>
</section>
`;
